src/DIRAC/MonitoringSystem/private/Plotters/DataOperationPlotter.py

```diff
@@ -0,0 +1,210 @@
+
+
+class DataOperationPlotter(BasePlotter):
+
+    """
+    .. class:: DataOperationPlotter
+
+    It is used to crate the plots.
+
+    param: str _typeName monitoring type
+    param: list _typeKeyFields list of keys what we monitor (list of attributes)
+    """
+
+    _typeName = "DataOperation"
+    _typeKeyFields = DataOperation().keyFields
+
+    def _translateGrouping(self, grouping):
+        if grouping == "Channel":
+            return ("%s, %s", ["Source", "Destination"], "CONCAT( %s, ' -> ', %s )")
+        else:
+            return ("%s", [grouping])
+
+    _reportSuceededTransfersName = "Successful transfers"
+
+    def _reportSuceededTransfers(self, reportRequest):
+        return self.__reportTransfers(reportRequest, "Succeeded", ("Failed", 0))
+
+    _reportFailedTransfersName = "Failed transfers"
+
+    def _reportFailedTransfers(self, reportRequest):
+        return self.__reportTransfers(reportRequest, "Failed", ("Succeeded", 1))
+
+    def __reportTransfers(self, reportRequest, titleType, togetherFieldsToPlot):
+        retVal = self._getTimedData(
+            reportRequest["startTime"],
+            reportRequest["endTime"],
+            "TransferTotal",
+            preCondDict=reportRequest["condDict"],
+            metadataDict=None,
+        )
+        if not retVal["OK"]:
+            return retVal
+        dataDict, granularity = retVal["Value"]
+
+        dataDict, maxValue = self._divideByFactor(dataDict, granularity)
+        dataDict = self._fillWithZero(granularity, reportRequest["startTime"], reportRequest["endTime"], dataDict)
+        baseDataDict, graphDataDict, maxValue, unitName = self._findSuitableRateUnit(
+            dataDict, self._getAccumulationMaxValue(dataDict), "files"
+        )
+        return S_OK(
+            {"data": baseDataDict, "graphDataDict": graphDataDict, "granularity": granularity, "unit": unitName}
+        )
+
+    def _plotSuceededTransfers(self, reportRequest, plotInfo, filename):
+        return self.__plotTransfers(reportRequest, plotInfo, filename, "Succeeded", ("Failed", 0))
+
+    def _plotFailedTransfers(self, reportRequest, plotInfo, filename):
+        return self.__plotTransfers(reportRequest, plotInfo, filename, "Failed", ("Succeeded", 1))
+
+    def __plotTransfers(self, reportRequest, plotInfo, filename, titleType, togetherFieldsToPlot):
+        metadata = {
+            "title": "%s Transfers by %s" % (titleType, reportRequest["grouping"]),
+            "ylabel": plotInfo["unit"],
+            "starttime": reportRequest["startTime"],
+            "endtime": reportRequest["endTime"],
+            "span": plotInfo["granularity"],
+        }
+        return self._generateTimedStackedBarPlot(filename, plotInfo["graphDataDict"], metadata)
+
+    _reportQualityName = "Efficiency by protocol"
+
+    def _reportQuality(self, reportRequest):
+        selectFields = (
+            "'Total', %s, %s, SUM(%s),SUM(%s)",
```

Comment:
These operations look like MySQL ones.

Reply:
You are right, I forgot to "translate" that function for ES. Fixing it right away.